### ISSUE TITLE
quic in chrome using android

### ISSUE THREAD

**Issue by r00tback** [mitmproxy/mitmproxy#7339]
#### Problem Description
when I try to use mitmproxy for quic in Android Chrome or any app i get tls handshake failed 
and i can't bypass this i tried Frida scripts etc but none Magisk added a certificate to root
all did't bypass this 
Only Firefox works if I make it trust a third-party certificate 
#### Steps to reproduce the behavior:
1. https://prnt.sc/cZh9A2HYkVRh

**Comment by errorxyz** [comment]
Can you provide more information about your setup? Which mode are you using? wireguard/reverse-proxy? Perhaps this is helpful: https://github.com/mitmproxy/mitmproxy/issues/7025#issuecomment-2351138170

**Comment by r00tback** [comment]
my setup nox player using wireguard mode
 firefox works if i make it trust thirdparty certificate 
 the issue in chrome even if i add certificate to root tls failed

**Comment by r00tback** [comment]
please its urgent if you know any method to make TLS work in Chrome in Android NOX player using wireguard or any method to make android trust the certificate for quic http3 ? 
all methods in doc's i tried for sure been trying for over 1 month to bypass this but none worked

**Comment by errorxyz** [comment]
mitmproxy does not support HTTP/3 over wireguard mode. HTTP/3 and VPNs generally don't work well together for some reason - sometimes it works and sometimes it doesn't. Since there's no consistent results we can't really pinpoint and fix the issue.

**Comment by r00tback** [comment]
but it work in firefox in android same setup nox player android wire guard mode
the issue is only in chrome or any app from google    On Saturday, November 23, 2024 at 07:50:32 AM GMT+3, Gaurav Jain ***@***.***> wrote:  
 
 


mitmproxy does not support HTTP/3 over wireguard mode. HTTP/3 and VPNs generally don't work well together for some reason - sometimes it works and sometimes it doesn't. Since there's no consistent results we can't really pinpoint and fix the issue.

**Comment by errorxyz** [comment]
HTTP/3 support for chrome on android is not supported. For chrome to support self signed certificates for HTTP/3 we'll need to manually specify a few command line switches (see the comment I'd linked earlier) but specifying these command line options is not possible on android.

**Comment by r00tback** [comment]
yes this exactly the issue android don't have this command line to allow self signed certificate 
and i need a way for it i tried ssl unpinning using Frida or other tools like Magisk modules but all didn't fix it
so i need way to do it in android 
    On Saturday, November 23, 2024 at 08:35:45 AM GMT+3, Gaurav Jain ***@***.***> wrote:  
 
 


HTTP/3 support for chrome on android is not supported. For chrome to support self signed certificates for HTTP/3 we'll need to manually specify a few command line switches (see the comment I'd linked earlier) but specifying these command line options is not possible on android.

**Comment by r00tback** [comment]
There is no workaround for this issue on Android Chrome or for universal injectors to use our self-signed certificates on chrome or any other app and http3!

**Comment by errorxyz** [comment]
possible helpful: https://www.chromium.org/developers/how-tos/run-chromium-with-flags/#android
